pkg/db-repo/repository.go

```diff
@@ -230,6 +240,15 @@ func (r *repository) Query(ctx context.Context, qb *QueryBuilder, result interfa
 	}
 
 	for i := range qb.where {
+		currentWhere := qb.where[i]
```

Comment:
Will this break any joins with conditions applied on the joined entity, e.g.
select* from user u join device d on u.device_id = d.id where d.type = 'phone'?

Maybe it also makes sense to add the check to the other methods as well to detect any cross-entity calls?

Reply:
No, it shouldn't break the Joins. since they are done separately.